### PR TITLE
Feature/ys dev add container params

### PR DESCRIPTION
**Pull request by YiranCdr**

### CODE REVIEW

File: src/cpac/api/authKey.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import uuid
+
+
+class AuthKey:
+    key = None
+
+    @staticmethod
+    def generateKey():
```
Comment (anibalsolon) [inline src/cpac/api/authKey.py:8]: In Python, method names are snake-case: generate_key

File: src/cpac/api/authKey.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        if not AuthKey.key:
+            AuthKey.key = str(uuid.uuid1())
+        return AuthKey.key
+
+    @staticmethod
+    def getKey():
```
Comment (anibalsolon) [inline src/cpac/api/authKey.py:14]: snake-case

File: src/cpac/api/backends/container.py [hunk 3]
```diff
@@ -225,6 +225,19 @@ async def run(self):
         if pipeline:
             command += ['--pipeline_file', pipeline]
 
+        if self.execution_params:
+            command += ['--pipeline_override',
+                        'maximumMemoryPerParticipant: %d'
+                        % int(self.execution_params['parallelPipeline'])] \
+                if 'parallelPipeline' in self.execution_params else []
```
Comment (anibalsolon) [inline src/cpac/api/backends/container.py:235]: Let's invert this logic for these three conditionals, like this:

```python
if 'parallelPipeline' in self.execution_params:
    parallel_pipelines = int(self.execution_params['parallelPipeline'])
    command += ['--pipeline_override', f'maximumMemoryPerParticipant: {parallel_pipelines}']
```

and it is not `maximumMemoryPerParticipant`, but `numParticipantsAtOnce`

File: src/cpac/api/server.py [hunk 4]
```diff
@@ -77,11 +78,16 @@ def get_argument(self, arg, default=None):
 
 
 class MainHandler(BaseHandler):
-    def get(self):
+    def post(self):
         scheduler = self.application.settings.get("scheduler")
         backend = scheduler.backend
         types = {v: k for k, v in available_backends.items()}
 
+        ifWrongAuthKey = False
```
Comment (anibalsolon) [inline src/cpac/api/server.py]: Please use snake case variables on Python

Comment (anibalsolon) [reply src/cpac/api/server.py]: and 

```python
wrong_key = 'authKey' not in self.json or self.json['authKey'].strip() != AuthKey.getKey()
```

**Review by anibalsolon — CHANGES_REQUESTED** [review]
The static solution to store the key is fine. However, your auth key solution does not prevent someone to access the other endpoints, nor the websocket.

The pipeline parameters looks good, just small logic changes.

File: src/cpac/api/authKey.py (hunk 1 above)
Comment (YiranCdr) [reply src/cpac/api/authKey.py:8]: Fixed.

**Pull request merged**